actioncable/test/subscription_adapter/common.rb

```diff
@@ -32,7 +33,7 @@ def subscribe_as_queue(channel, adapter = @rx_adapter)
     subscribed.wait(WAIT_WHEN_EXPECTING_EVENT)
     assert subscribed.set?
 
-    yield queue
+    Timeout.timeout(WAIT_WHEN_EXPECTING_EVENT) { yield queue }
```

Comment:
Could you, please, clarify?

That's an easiest way to handle blocking `queue.pop` calls in these tests (which cause a process to stuck forever).

Of course, it would be better to refactor the tests to not use blocking calls though.

Another idea is to "enhance" `Queue#pop` with built-in timeout through refinements:

```ruby
refine Queue do
  def pop(timeout)
    while empty?
      sleep 0.1
      timeout -= 0.1
      return nil unless timeout > 0
    end
    super
  end
end
```

Or we can add custom queue class.

But either way, I think, we should be able to make tests fail (which is impossible now).